shared/devices/index.js

```diff
@@ -35,18 +38,28 @@ class Devices extends React.PureComponent<Props, State> {
     this.props.loadDevices()
   }
 
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    if (this.props.hasNewlyRevoked && !prevState.revokedExpanded) {
+      this.setState({revokedExpanded: true})
+    }
+  }
+
   _toggleExpanded = () => this.setState(p => ({revokedExpanded: !p.revokedExpanded}))
 
-  _renderRow = (index, item) =>
-    item.type === 'revokedHeader' ? (
-      <RevokedHeader
-        key="revokedHeader"
-        expanded={this.state.revokedExpanded}
-        onToggleExpanded={this._toggleExpanded}
-      />
-    ) : (
-      <DeviceRow key={item.id} deviceID={item.id} firstItem={index === 0} />
-    )
+  _renderRow = (index, item) => {
+    if (item.type === 'revokedHeader') {
+      return (
+        <RevokedHeader
+          key="revokedHeader"
+          expanded={this.state.revokedExpanded}
+          onToggleExpanded={this._toggleExpanded}
+        />
+      )
+    } else {
+      const newItem = this.props.newlyChangedItemIds.has(item.id)
+      return <DeviceRow key={item.id} deviceID={item.id} firstItem={index === 0} isNew={newItem} />
```

Comment:
so the common pattern is to not plumb through this kind of thing to the rows. instead the row has its own connector and just just pull its own state out itself

Reply:
oh. ok. i can do that.